### ISSUE TITLE
Regression in 1.1.239 for TypeVar bound to Literal

### ISSUE THREAD

**Issue by RobertCraigie** [microsoft/pyright#3366]
**Describe the bug**

```py
from typing import Literal, TypeVar, Mapping

SortOrder = Literal['asc', 'desc']
ScalarKeysT = TypeVar('ScalarKeysT', bound=Literal['name', 'country'])

def group_by(
    by: list[ScalarKeysT],
    *,
    order: Mapping[ScalarKeysT, SortOrder] | None = None,
) -> None:
    ...

group_by(
    ['country'],  # error: list[str] cannot be assigned to parameter "by" of type "list[ScalarKeysT@group_by]"
    order={  # error: dict[str, str] cannot be assigned to parameter "order" of type "Mapping[ScalarKeysT@group_by, SortOrder] | None
        'country': 'asc',
    },
)
```

The above code passed type checks prior to `v1.1.239` and correctly enforced the inferred bound Literal such that the following call would correctly report an error as the `name` key is not present in the `by` argument:

```py
group_by(
    ['country'],
    order={  # error
        'name': 'asc',
    },
)
```

**Expected behavior**

The first code sample provided should pass type checks.

**VS Code extension or command-line**

Command-line `v1.1.239`

**Comment by erictraut** [comment]
Thanks for the bug report.

I've narrowed down the regression to this commit: https://github.com/microsoft/pyright/commit/f5d97f3341b1d36ab8b64544c4078f3b9046d7b3. I'll need to dig into it in more detail to fully diagnose.

FWIW, mypy also emits an error for this code too, but I think that's a bug.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.243, which I just published. It will also be included in the next release of pylance.
